### PR TITLE
fix(logs): Updated remaining log messages to match standards

### LINKED ISSUE #2289: Unify log message style
We should make a pass through our logging statements and ensure that they all match a consistent style. They should not be capitalized, their wording should be concise, and they should use `tracing`'s k/v syntax instead of string formatting.

There is a bit of an open question about whether messages should end with a period. @LucioFranco thinks it looks better, but I'm somewhat skeptical. Please weigh in if you feel strongly, otherwise we can flip a coin.

### PR DESCRIPTION
**Pull request by jamtur01**
Closes #2289 

Standards:

1. Always use Tracing's key/value style for log events.
2. Never use `e` or `err` - always spell out `error` to enrich logs and make it clear what the output is.
3. Prefer Display over Debug, `?error` and not `%error`.

Nope!

```
warn!("Failed to merge value: {}.", err);
```

Yep!

```
warn!(message = "Failed to merge value", error = ?error);
```

Signed-off-by: James Turnbull <james@lovedthanlost.net>

### CODE REVIEW

**Comment by jamtur01** [comment]
c.f. #3850.

**Review by fanatid — APPROVED** [review]

**Review by jszwedko — COMMENTED** [review]
Thanks for taking this on :smile: 

We could decide to split this off, but one part of the referenced issue, https://github.com/timberio/vector/issues/2289, was preferring the key/value format over formatting a string.

That is, rather than:

```
                                 warn!("Failed to merge value: {}.", err);
```

Prefer:

```
                                 warn!("Failed to merge value.", error = %err);
```

Not related to this PR, but I noticed we are using informal names for components in some of the messages (like '"Fatal cloudwatchlogs sink error: {}.` rather than `"Fatal aws_cloudwatch_logs sink error: {}.`. It might be nice to standardize on using the external component name.

I started to leave some comments for each message that needed to be updated to use the key/value format but there were a lot of them that need updated :smile:

**Comment by jszwedko** [comment]
As a follow-up, I think it'd be nice to extend the check @bruceg added for internal events to cover this case as well: https://github.com/timberio/vector/blob/master/scripts/check-internal-events.sh

That should help reduce re-introductions of ill-formated messages.

**Comment by jamtur01** [comment]
@jszwedko That's how I found most of these - I need to tweak the regex though as there are a lot of false positives.

**Review by bruceg — COMMENTED** [review]
This looks good, but I agree with @jszwedko, we should at least add/modify a script to check this doesn't creep back, and probably go all the way and standardize on key/value instead of format strings.

**Comment by FungusHumungus** [comment]
Additionally, it would be good to add these standards to the [Contributing.md](https://github.com/timberio/vector/blob/master/CONTRIBUTING.md).

File: src/cli.rs [hunk 1]
```diff
@@ -169,7 +169,7 @@ impl std::str::FromStr for LogFormat {
 
 pub fn handle_config_errors(errors: Vec<String>) -> exitcode::ExitCode {
     for error in errors {
-        error!("Configuration error: {}", error);
+        error!("Configuration error: {}.", error);
```
[the remark this answers is not in the document]
Comment (jamtur01) [reply src/cli.rs:172]: Hmmm I don't think this works:

```
error: expected expression, found `%`
   --> src/cli.rs:172:40
    |
172 |         error!("Configuration error.", %error);
    |                                        ^ expected expression
```

[the remark this answers is not in the document]
Comment (jszwedko) [reply src/cli.rs:172]: Ah, my fault, I forgot the syntax has to be different when using `%variable`:

```
         error!(message = "Configuration error.", %error);
```

File: src/topology/mod.rs [hunk 2]
```diff
@@ -143,7 +143,7 @@ impl RunningTopology {
             let remaining_components = check_handles2.keys().cloned().collect::<Vec<_>>();
 
             error!(
-                "Failed to gracefully shut down in time. Killing: {}",
+                "Failed to gracefully shut down in time. Killing: {}.",
                 remaining_components.join(", ")
```
Comment (bruceg) [inline src/topology/mod.rs:147]: Would this make sense?
```suggestion
                "Failed to gracefully shut down in time. Killing components.",
                components = %remaining_components
```

File: src/transforms/reduce/mod.rs [hunk 3]
```diff
@@ -87,7 +87,7 @@ impl ReduceState {
                         match get_value_merger(v, strat) {
                             Ok(m) => Some((k, m)),
                             Err(err) => {
-                                warn!("failed to create merger for field '{}': {}", k, err);
+                                warn!("Failed to create merger for field '{}': {}.", k, err);
```
Comment (bruceg) [inline src/transforms/reduce/mod.rs:90]:
```suggestion
                                warn!("Failed to create merger.", field = ?k, ?err);
```

File: tests/support/mod.rs [hunk 4]
```diff
@@ -168,8 +168,8 @@ impl SourceConfig for MockSourceConfig {
                 }
                 x
             })
-            .forward(out.sink_map_err(|e| error!("Error sending in sink {}", e)))
-            .map(|_| info!("Finished sending"))
+            .forward(out.sink_map_err(|e| error!("Error sending in sink {}.", e)))
```
Comment (bruceg) [inline tests/support/mod.rs:171]:
```suggestion
            .forward(out.sink_map_err(|error| error!("Error sending in sink.", ?error)))
```

**Review by bruceg — COMMENTED** [review]
I appreciate regularizing on Display instead of Debug for errors. There were several missed in this go around (I didn't flag them all) and I suggested a few other tweaks.

File: src/conditions/check_fields.rs [hunk 5]
```diff
@@ -450,7 +450,7 @@ fn build_predicate(
         "contains" => ContainsPredicate::new(target, arg),
         "prefix" => {
             warn!(
-                message = "The \"prefix\" comparison predicate is deprecated, use \"starts_with\" instead",
+                message = "The 'prefix' comparison predicate is deprecated, use 'starts_with' instead.",
```
[the remark this answers is not in the document]
Comment (Hoverbear) [reply src/conditions/check_fields.rs:453]: Oh heck I was going to suggest `

[the remark this answers is not in the document]
Comment (jamtur01) [reply src/conditions/check_fields.rs:453]: Oh that's what I meant to do. Will fix.

**Review by binarylogic — APPROVED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -429,6 +429,25 @@ rustup component add rustfmt
...
+Nope!
+
+```rust
+warn!("Failed to merge value: {}.", err);
+```
+
+Yep!
+
+```rust
+warn!(message = "Failed to merge value", error = ?error);
```
Comment (FungusHumungus) [inline CONTRIBUTING.md:448]:
```suggestion
warn!(message = "Failed to merge value.", error = ?error);
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -429,6 +429,25 @@ rustup component add rustfmt
 make fmt
 ```
 
+##### Logging style
+
+
+- Always use Tracing's key/value style for log events.
+- Never use `e` or `err` - always spell out `error` to enrich logs and make it clear what the output is.
+- Prefer Display over Debug, `?error` and not `%error`.
+
```
Comment (FungusHumungus) [inline CONTRIBUTING.md:438]: Do we also want to say: 

- Capitalize and punctuate the message.

**Review by bruceg — APPROVED** [review]

**Comment by JeanMertz** [comment]
Thanks for all these changes @jamtur01.

I wanted to leave a (post-merge) note that I believe there's a difference between applying these styles to the _internal events_, versus applying them to the _internal code_.

That is to say, I see changes like these in the PR:

```diff
- .map_err(|e| e.to_string())
+ .map_err(|error| error.to_string())
```

This has no impact on internal events, and while I agree in general that clearly named variables is important, there are situations (such as here) where more verbosity doesn't add any clarity (and might in fact reduce clarity due to `rustfmt` line-length restrictions).

Similarly, a change such as this one:

```diff
- error!(message = "Unable to start config watcher.", %error);
+ error!(message = "Unable to start config watcher.", error = ?error);
```

I don't see a problem in using short-hand field notations, as it's something every Rustacean is very familiar with, in the context of struct initialization.

Perhaps some of these changes were applied through the magic of `sed`, but I just wanted to make sure we're aligned on these patterns applying to the _representation_ of internal event, not how we construct those events (or Rust source code in general).

**Comment by jszwedko** [comment]
@JeanMertz just noting that I switched to short-hand notation in https://github.com/timberio/vector/pull/4790#issuecomment-717939215

**Comment by jamtur01** [comment]
@JeanMertz We're aligned. I prefer the explicit and I changed a few where I felt it was not but not deeply invested.

**Pull request merged**